src/rest/authFetch.js

```diff
@@ -0,0 +1,35 @@
+import fetch from 'node-fetch'
+import debugFactory from 'debug'
+import AuthFetchError from '../errors/AuthFetchError'
+
+const debug = debugFactory('StreamrClient:utils')
+
+const authFetch = async (url, session, opts = {}, requireNewToken = false) => {
+    debug('authFetch: ', url, opts)
+
+    const response = await fetch(url, {
+        ...opts,
+        headers: {
+            ...(session && !session.options.unauthenticated ? {
+                Authorization: `Bearer ${await session.getSessionToken(requireNewToken)}`,
+            } : {}),
+            ...opts.headers,
+        },
+    })
+
+    const body = await response.text()
+
+    if (response.ok) {
+        try {
+            return JSON.parse(body || '{}')
+        } catch (e) {
+            throw new AuthFetchError(`Failed to parse JSON response: ${body}`, response, body)
```

Comment:
This is inherited from the existing code but consider the case when `body` is a massive chunk of html, the error message will be pretty unusable. Perhaps should just leave it out of the message entirely since we're attaching to the error object, or limit the damage with `body.slice(0, 100)`.

Reply:
Also possible this could re-use the existing JSON parse message `throw new AuthFetchError(e.message, response, body)`